# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/GLImages.cpp
# Commit: c333871e1c59
# Full Hash: c333871e1c59abe6daf30c396e5e0a09591ac440
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2022-01-19 09:34:35
# Regressor Bug: 1640607
# File Overlap Count: 1
# Description:
#   Bug 1736793 - Add SurfaceTexture support to GLBlitHelper::BlitSdToFramebuffer() r=jgilbert,gfx-reviewers
#   
#   Blit support of android SurfaceTexture was removed by Bug 1640607.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D128992
# ==============================================================================

diff -r bb40a646b246 -r c333871e1c59 gfx/layers/GLImages.cpp
--- a/gfx/layers/GLImages.cpp	Wed Jan 19 02:03:31 2022 +0000
+++ b/gfx/layers/GLImages.cpp	Wed Jan 19 02:13:48 2022 +0000
@@ -85,6 +85,14 @@
       mHasAlpha(aHasAlpha) {
   MOZ_ASSERT(mHandle);
 }
+
+Maybe<SurfaceDescriptor> SurfaceTextureImage::GetDesc() {
+  SurfaceDescriptor sd = SurfaceTextureDescriptor(
+      mHandle, mSize,
+      mHasAlpha ? gfx::SurfaceFormat::R8G8B8A8 : gfx::SurfaceFormat::R8G8B8X8,
+      false /* NOT continuous */, false /* do not ignore transform */);
+  return Some(sd);
+}
 #endif
 
 }  // namespace layers